### ISSUE TITLE
Fix extra nonzero call for chained comparisons

### ISSUE THREAD

**Issue by kmod** [dropbox/pyston#1190]
See [compare_ordering.py](https://github.com/dropbox/pyston/blob/master/test/tests/compare_ordering.py)

The issue is that for chanined comparisons (`a < b < c`), we do something like:
```
def chained_compare(a, b, c):
  t1 = a < b
  if not bool(t1):
    return t1
  else:
    t2 = b < c
    nonzero(t2)
    return t2
```

Where that final `nonzero(t2)` call is extraneous and should be removed.

**Comment by benhoyt** [comment]
@kmod FYI, that link is broken. Should it be [here](https://github.com/dropbox/pyston/blob/master/test/tests/compare_ordering.py)?

**Comment by kmod** [comment]
Oops yes, thanks :)

**Comment by sizeoftank** [comment]
Hi  @kmod 
I found your comment here  when  I debugging for this issue.
If so can we add a check to skip the last branch here?
````
    // TODO: this branch is unused for the last comparison
    AST_Branch* br = new AST_Branch();
    br->test = callNonzero(makeLoad(name, node));
    push_back(br);
    
    CFGBlock* was_block = curblock;
    CFGBlock* next_block = cfg->addBlock();
    CFGBlock* crit_break_block = cfg->addBlock();
    was_block->connectTo(next_block);
    was_block->connectTo(crit_break_block);
    
    br->iffalse = crit_break_block;
    br->iftrue = next_block;
    
    curblock = crit_break_block;
    pushJump(exit_block);
    
    curblock = next_block;
    
    left = right;
````

**Comment by kmod** [comment]
Closed via #1301
